patroni/psycopg.py

```diff
@@ -14,27 +22,79 @@
+
+        :param value: value to be quoted.
+        :param conn: if a connection is given then :func:`quote_literal` checks if any special handling based on server
+            parameters needs to be applied to *value* before quoting it as a SQL literal.
+
+        :returns: *value* quoted as a SQL literal.
+        """
         value = adapt(value)
         if conn:
             value.prepare(conn)
         return value.getquoted().decode('utf-8')
 except ImportError:
-    from psycopg import connect as __connect, sql, Error, DatabaseError, OperationalError, ProgrammingError
+    from psycopg import connect as __connect, sql, Error, DatabaseError, OperationalError, ProgrammingError, Connection
+
+    def _connect(*args: Any, **kwargs: Any) -> Connection:
+        """Call ``psycopg.connect`` with ``*args`` and ``**kwargs``.
+
+        .. note::
+            Will create ``server_version`` attribute in the returning connection, so it keeps compatibility with the
+            object that would be returned by ``psycopg2.connect``.
+
+        :param args: positional arguments to call ``psycopg.connect`` with.
+        :param kwargs: keyword arguments to call ``psycopg.connect`` with.
 
-    def _connect(*args, **kwargs):
+        :returns: a connection to the database.
+        """
         ret = __connect(*args, **kwargs)
         ret.server_version = ret.pgconn.server_version  # compatibility with psycopg2
         return ret
 
-    def _quote_ident(value, conn):
+    def _quote_ident(value: Any, conn: Connection) -> str:
+        """Quote *value* as a SQL identifier.
+
+        :param value: value to be quoted.
+        :param conn: connection to evaluate the returning string into.
+
+        :returns: *value* quoted as a SQL identifier.
+        """
         return sql.Identifier(value).as_string(conn)
 
-    def quote_literal(value, conn=None):
+    def quote_literal(value: Any, conn: Optional[Connection] = None) -> str:
+        """Quote *value* as a SQL literal.
+
+        :param value: value to be quoted.
+        :param conn: connection to evaluate the returning string into.
+
+        :returns: *value* quoted as a SQL literal.
+        """
         return sql.Literal(value).as_string(conn)
 
 
-def connect(*args, **kwargs):
+def connect(*args: Any, **kwargs: Any) -> Any:
+    """Get a connection to the database.
+
+    .. note::
+        The connection will have ``autocommit`` enabled.
+
+    :param args: positional arguments to call ``connect`` function from ``psycopg`` module.
+    :param kwargs: keyword arguments to call ``connect`` function from ``psycopg`` module.
+
+    :returns: a connection to the database. Can be either a :class:`psycopg.Connection` if using ``psycopg3``, or a
+        :class:`psycopg2.extensions.connection` if using ``psycopg2``.
+    """
+    # Enforce search_path=pg_catalog for non-replication connections to mitigate security issues as Patroni relies on
```

Comment:
Should this comment be a part of docstring?

Reply:
I think that makes sense!